packages/turbo-workspace-convert/src/utils.ts

```diff
@@ -0,0 +1,239 @@
+}): Array<string> {
+  const workspaceFile = path.join(workspaceRoot, "pnpm-workspace.yaml");
+  if (fs.existsSync(workspaceFile)) {
+    const workspaceConfig = yaml.load(
+      fs.readFileSync(workspaceFile, "utf8")
+    ) as { packages: Array<string> };
+    return workspaceConfig.packages;
+  }
+
+  return [];
+}
+
+function expandWorkspaces({
+  workspaceRoot,
+  workspaceGlobs,
+}: {
+  workspaceRoot: string;
+  workspaceGlobs?: string[];
+}): Array<Workspace> {
+  if (!workspaceGlobs) {
+    return [];
+  }
+  return workspaceGlobs
+    .flatMap((workspaceGlob) => {
+      const workspacePackageJsonGlob = `${workspaceGlob}/package.json`;
+      return glob.sync(workspacePackageJsonGlob, {
+        onlyFiles: true,
+        absolute: true,
+        cwd: workspaceRoot,
+      });
+    })
+    .map((workspacePackageJson) => {
+      const workspaceRoot = path.dirname(workspacePackageJson);
+      const name = getWorkspaceName({ workspaceRoot });
+      return {
+        name,
+        paths: {
+          root: workspaceRoot,
+          packageJson: workspacePackageJson,
+          nodeModules: path.join(workspaceRoot, "node_modules"),
+        },
+      };
+    });
+}
+
+function updateDependencyList({
+  dependencyList,
+  project,
+  to,
+}: {
+  dependencyList: DependencyList;
+  project: Project;
+  to: PackageManagerDetails;
+}): { dependencyList: DependencyList; updated: Array<string> } {
+  const updated: Array<string> = [];
+  project.workspaceData.workspaces.forEach((workspace) => {
+    const { name } = workspace;
+    if (dependencyList[name]) {
+      dependencyList[name] = to.name === "pnpm" ? `workspace:*` : "*";
+      updated.push(name);
+    }
+  });
+
+  return { dependencyList, updated };
+}
+
+function updateDependencies({
+  project,
+  workspace,
+  to,
+  logger,
+  options,
+}: {
+  workspace: Workspace;
+  project: Project;
+  to: PackageManagerDetails;
+  logger: Logger;
+  options?: Options;
+}) {
+  const workspacePackageJson = getPackageJson({
+    workspaceRoot: workspace.paths.root,
+  });
+
+  // update all dependencies
+  const stats: Record<string, Array<string>> = {
+    dependencies: [],
+    devDependencies: [],
+  };
+  const { devDependencies, dependencies } = workspacePackageJson;
+  if (devDependencies) {
+    const { updated, dependencyList } = updateDependencyList({
+      dependencyList: devDependencies,
+      project,
+      to,
+    });
+    workspacePackageJson["devDependencies"] = dependencyList;
+    stats.devDependencies = updated;
+  }
+  if (dependencies) {
+    const { updated, dependencyList } = updateDependencyList({
+      dependencyList: dependencies,
+      project,
+      to,
+    });
+    workspacePackageJson["dependencies"] = dependencyList;
+    stats.dependencies = updated;
+  }
+
+  const numDevDependencies = stats.devDependencies.length;
+  const numDependencies = stats.dependencies.length;
+  const suffix = (num: number) => `${num <= 1 ? "y" : "ies"}`;
+
+  // both updated
```

Comment:
This whole `if`/`else if` strategy needs to change to allow for `optionalDependencies` and `peerDependencies`